pkg/prometheus/promcfg_test.go

```diff
@@ -1581,3 +1587,53 @@ func TestHonorLabels(t *testing.T) {
 		}
 	}
 }
+
+func TestHonorTimestamps(t *testing.T) {
+	type testCase struct {
+		UserHonorTimestamps     *bool
+		OverrideHonorTimestamps bool
+		Expected                bool
+	}
+
+	no := false
+	yes := true
+	testCases := []testCase{
+		{
+			UserHonorTimestamps:     nil,
```

Comment:
`User` means here `serviceMonitor` honor timestamp?

Reply:
Can we be explicit?

Reply:
This is a convention from testing honor_labels, I didn't want to diverge.

Reply:
> User means here serviceMonitor honor timestamp?

ServiceMonitor or PodMonitor